hw_2/fastq_filtrator.py

```diff
@@ -0,0 +1,63 @@
+# global filtering function
+def main(input_fastq, output_file_prefix, gc_bounds=(20, 80),
+         length_bounds=(0, 2 ** 32), quality_threshold=0, save_filtered=False):
+
+    # dictionary for positive and negative selection of reads
+    dict_selection = {0: [], 1: []}
+
+    # create additional variable; it's allow to not read entire file
+    i = 0
+    sep_line = []
+
+    # filtering function according to GC-content of sequence
+    def gc_count(sequence, gc_broad=gc_bounds):
+        if type(gc_broad) != tuple:                          # in case one variable
+            gc_broad = (0, gc_broad)
+        min_gc, max_gc = gc_broad
+        gc_percent = (sequence.count('G')+sequence.count('C'))*100/len(sequence)
+        if min_gc <= gc_percent <= max_gc:
+            return 1
+        else:
+            return 0
```

Comment:
Можно сильно упростить
```suggestion
        return min_gc <= gc_percent <= max_gc:
```